unblob/pool.py

```diff
@@ -0,0 +1,118 @@
+import abc
+import multiprocessing as mp
+import sys
+from multiprocessing.queues import JoinableQueue
+from typing import Any, Callable, Union
+
+from .logging import multiprocessing_breakpoint
+
+
+class PoolBase(abc.ABC):
+    @abc.abstractmethod
+    def submit(self, args):
+        pass
+
+    @abc.abstractmethod
+    def process_until_done(self):
+        pass
+
+    def start(self):
+        pass
+
+    def close(self):
+        pass
+
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+
+
+class Queue(JoinableQueue):
+    def is_empty(self) -> bool:
+        """Checks if all ``task_done`` has been called for all items.
+        Based on ``multiprocessing.JoinableQueue.join``."""
+        with self._cond:  # type: ignore
+            return self._unfinished_tasks._semlock._is_zero()  # type: ignore
+
+
+def _worker_process(handler, input, output):
+    sys.breakpointhook = multiprocessing_breakpoint
+    while True:
+        args = input.get()
+        result = handler(args)
+        output.put(result)
+
+
+class MultiPool(PoolBase):
+    def __init__(
+        self,
+        process_num: int,
+        handler: Callable[[Any], Any],
+        *,
+        result_callback: Callable[["MultiPool", Any], Any],
+    ):
+        if process_num <= 0:
+            raise ValueError("At process_num must be greater than 0!")
```

Comment:
I don't think that it worth to use this class with zero processes :)

Reply:
Yes... hence the `ValueError` :sweat_smile: 

Should I those conditions and let the "user" (us....basically....) hand off these restrictions: